extensions/reclaim-ai/src/utils/events.ts

```diff
@@ -23,3 +23,58 @@ export const truncateEventSize = (eventTitle: string) => {
   }
   return eventTitle;
 };
+
+const CalendarSyncTitleRegex = /(Personal|Work|Travel) Commitment$/;
+
+/**
+ * Group events based on their start-end time combination.
+ * @param events List of events to group.
+ * @returns Map of grouped events.
+ */
+const groupEventsByTime = (events: ApiResponseEvents) => {
+  return events.reduce((acc, event) => {
+    const key = `${event.eventStart}-${event.eventEnd}` as const;
+    if (acc.has(key)) acc.get(key)!.push(event);
+    else acc.set(key, [event]);
+    return acc;
+  }, new Map<string, ApiResponseEvents>());
+};
+
+/**
+ * Filter events based on the given criteria.
+ * @param events List of events to filter.
+ * @returns Filtered list of events.
+ */
+const filterRelevantEvents = (events: ApiResponseEvents) => {
+  return events.filter((event) => {
+    const isSyncedAndManaged = event.personalSync && event.reclaimManaged;
+    const matchesTitleCriteria =
```

Comment:
Talked with @lightbody about this one, we have an internal (shared here) doc:

https://docs.google.com/document/d/1Hc2d1BE9-Er6-UUoD6v7eZRAJGCigiVHOcotnSuZgCs/edit?usp=sharing

that details, I think, how you could very very reliably detect / link sync events to the source event, without needing to look at the title and the start / end time. Maybe take that doc for a spin and let me know if you think that would simplify things?

Reply:
Thanks for the information! It might indeed be a better way @shkreios 🙂

Reply:
@ajohnson1200, I greatly appreciate your input. I've refined the filtering logic based on your suggestions. However, I'm not entirely confident that my updated implementation accurately addresses the corner cases of recurring events. Could you, or perhaps @lightbody, spare a moment to review it? Thank you!